usr/src/cmd/tip/value.c

```diff
@@ -152,7 +152,7 @@ vtoken(char *s)
 		if (p = vlookup(s)) {
 			cp++;
 			if (p->v_type&NUMBER)
-				vassign(p, (char *)atoi(cp));
+				vassign(p, (char *)(intptr_t)atoi(cp));
```

Comment:
This is truly awful and ideally vassign() needs rewriting and the zzhack stuff ripping out, but this is no worse than what's already there and gets it building 64-bit, so..